lib/parsers/package-lock-parser.ts

```diff
@@ -49,66 +64,234 @@ export class PackageLockParser implements LockfileParser {
+    cycle: string[], depMap: DepMap, depGraph: graphlib.Graph): {[cycleStart: string]: string} {
+
+    function walkCycleRec(node, traversed, currentCycle, nodeCopy) {
+      // 2. For every cyclic dependency of entry node...
+      const edgesToProcess = (depGraph.inEdges(node) as graphlib.Edge[])
+        .filter((e) => currentCycle.includes(e.v));
+      for (const edge of edgesToProcess) {
+      // ... create a duplicate of the dependency...
+        const copyName = edge.v + uuid();
+      // ...and connect it with the duplicated entry node
+        depGraph.setEdge(copyName, nodeCopy);
+        // metadata update
+        depMap[copyName] = _.cloneDeep(depMap[edge.v]);
+        if (traversed.includes(edge.v)) {
+          // 3.a If edge goes to already-visited dependency, end of cycle is found;
+          // update metadata and do not continue traversing
+          depMap[copyName].cyclic = true;
+        } else {
+          // 3.b Follow the edge and repeat the process, storing visited dependency-paths
+          // metadata update
+          walkCycleRec(edge.v, [...traversed, node], currentCycle, copyName);
+          // All non-cyclic dependencies of duplicated node need to be updated.
+          const edgesToCopy = (depGraph.inEdges(edge.v) as graphlib.Edge[])
+            .filter((e) => !currentCycle.includes(e.v));
+          for (const edgeToCopy of edgesToCopy) {
+            depGraph.setEdge(edgeToCopy.v, copyName);
+          }
+        }
+      }
+    }
+
+    const cycleStarts = {};
+    // For every node in a cycle:
+    for (const start of cycle) {
+      // 1. Create a uniqe duplicate of entry node (without edges)
+      const newNode = start + uuid();
+      cycleStarts[start] = newNode;
+      depMap[newNode] = depMap[start];
+      // update depMap with new node
+      depGraph.setNode(newNode);
+      walkCycleRec(start, [], cycle, newNode);
+      // 4. All non-cyclic dependencies or dependants of original node need to be
+      //   updated to be connected with the duplicated one
+
+      // dependants
+      const edges = depGraph.nodeEdges(start) as graphlib.Edge[]; // node has to have edges
+      const entryEdges = edges.filter((e) => !_.includes(cycle, e.w));
+      for (const edge of entryEdges) {
+        depGraph.setEdge(newNode, edge.w);
+      }
+      // dependencies
+      const outEdges = edges.filter((e) => !_.includes(cycle, e.v));
+      for (const edge of outEdges) {
+        depGraph.setEdge(edge.v, newNode);
+      }
+    }
+
+    // Once completed for all nodes in a cycle, original cyclic nodes can
+    // be removed.
+    for (const start of cycle) {
+      depGraph.removeNode(start);
+    }
+
+    return cycleStarts;
+  }
+
+  private createGraphOfDependencies(depMap: DepMap): graphlib.Graph {
+    const depGraph = new graphlib.Graph();
+    for (const depName of Object.keys(depMap)) {
+      depGraph.setNode(depName, depName);
+    }
+    for (const [depPath, dep] of _.entries(depMap)) {
```

Comment:
`_.entries(` -> `Object.entries(`?
If you don't mind, I prefere to use native JS features whenever it's possible instead of lodash.